nevergrad/functions/perfcap3d/core.py

```diff
@@ -0,0 +1,436 @@
+import os
+import pathlib
+import typing as tp
+import logging
+import itertools
+import copy
+from enum import Enum
+from uuid import uuid4 as uuid_gen
+from nevergrad.parametrization import parameter as p
+from .rmqservice import RMQService as _RMQService, RMQSettings as _RMQSettings
+from ...functions.base import ExperimentFunction
+
+class Status(Enum):
+    OK = 0
+    Failed = 1
+
+class VarNormalizer:
+    """
+    Normalizes parameter values to range [-1.0,1.0] based on bounds
+    This may improve optimizer convergence when scaling is different across parameters (i.e root joint translation vs joints rotation)
+    Can be silently deactivated if normalize is set to False in the constructor.
+    """
+
+    def __init__(self, lower_bound: float, upper_bound: float, normalize: bool = False):
+        """
+        lower_bound: variable lower bound
+        upper_bound: variable upper bound
+        normalize: if False VarNormalizer is deactivated. If True, it will normalize input values in [-1.0,1.0]
+        """
+        self._lower_bound = lower_bound
+        self._upper_bound = upper_bound
+        self._normalize = normalize
+
+    def denormalize(self, val: float) -> float:
+        """
+        Denormalize the parameter val
+        val: value to denormalize
+
+        Returns denormalized value
+        """
+        if not self._normalize:
+            return val
+        else:
+            return (val+1)*(self._upper_bound-self._lower_bound)/2.0 + self._lower_bound
+
+    def normalize(self, val: float) -> float:
+        """
+        Normalize the parameter val
+        val: value to normalize
+
+        Returns normalized value
+        """
+        if not self._normalize:
+            return val
+        else:
+            return 2.0 * (val - self.lower_bound) / (self._upper_bound - self._lower_bound) - 1.0
+
+    @property
+    def lower_bound(self):
+        return self._lower_bound
+
+    @property
+    def upper_bound(self):
+        return self._upper_bound
+
+    @property
+    def active_lower_bound(self):
+        """
+        effective lower variable bound
+        """
+        if self._normalize:
+            return -1.0
+        else:
+            return self._lower_bound
+
+    @property
+    def active_upper_bound(self):
+        """
+        effective upper variable bound
+        """
+
+        if self._normalize:
+            return 1.0
+        else:
+            return self._upper_bound
+
+    @property
+    def active(self):
+        """
+        Returns True, if VarNormalizer is active
+        """
+        return self._normalize
+
+# pylint:disable=too-many-instance-attributes
+
+class Perfcap3DFunction(ExperimentFunction):
+    """
+    Performance Capture Objective Function. This appears as a blocking call to the caller.
+    However, under the hood, it does a network request via Perfcap3DServerComm's RMQService to
+    the Perfcap Benchmark Server to get the function evaluation at the specified point
```

Comment:
It there are stuff to install to make this work, the docstring should explain it.
If the download of the executable could be automatized, it could be even better

Reply:
I added automated downloading and unzipping of executable and data

Reply:
I added automated downloading, unzipping of executable and data, and execution of server on benchmark start